### PR TITLE
[CDAP-21059] introduce error category class

### PR DESCRIPTION
**Pull request by itsankit-google**

### CODE REVIEW

File: cdap-api-common/src/main/java/io/cdap/cdap/api/exception/ErrorCategory.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+ *
+ * <p>This class is used to classify errors into different categories,
+ * such as plugin errors, provisioning errors, etc.
+ * </p>
+ */
+public class ErrorCategory {
+  public static final String PLUGIN = "Plugin";
+  public static final String OTHERS = "Others";
+  private final boolean isPlugin;
+  private final String errorCategory;
```
Comment (adrikagupta) [inline cdap-api-common/src/main/java/io/cdap/cdap/api/exception/ErrorCategory.java]: When we say errorCategory don't we actually mean error sub category?  Looking at this variable it feels that it stores the whole category like "plugin - label" but it doesn't. Should we rename it or can we please add a comment to explain it?

Comment (itsankit-google) [reply cdap-api-common/src/main/java/io/cdap/cdap/api/exception/ErrorCategory.java]: It is not necessarily error sub category, it does stores the whole category like `Plugin-'<STAGE_NAME>'` but if for some reason we don't have the stage name right now it just stores `Plugin`.

File: cdap-api-common/src/main/java/io/cdap/cdap/api/exception/ErrorCategory.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  public ErrorCategory(boolean isPlugin) {
+    this.isPlugin = isPlugin;
+    this.errorCategory = null;
+  }
+
+  /*
+   * Returns the category of the error.
+   */
+  public String getErrorCategory() {
+    return errorCategory == null ? (isPlugin ? PLUGIN : OTHERS) :
```
Comment (sahusanket) [inline cdap-api-common/src/main/java/io/cdap/cdap/api/exception/ErrorCategory.java]: Are we going with 2 categories only ? 
if it's more, we should have an Enum ?

Comment (itsankit-google) [reply cdap-api-common/src/main/java/io/cdap/cdap/api/exception/ErrorCategory.java]: Introduced an error category `enum` and sub category as `string`.

Comment (itsankit-google) [reply cdap-api-common/src/main/java/io/cdap/cdap/api/exception/ErrorCategory.java]: `ErrorCategoryEnum` can be extended as per requirements.

**Review by sahusanket — APPROVED** [review]

**Pull request merged**